pkg/kubelet/kubelet.go

```diff
@@ -901,6 +901,23 @@ func (kl *Kubelet) initialNodeStatus() (*api.Node, error) {
 		if err != nil {
 			return nil, err
 		}
+
+		// If the cloud has zone information, label the node with the zone information
+		zones, ok := kl.cloud.Zones()
```

Comment:
I agree that letting this in, with the intention that I will refactor it out into a future plugin system makes a lot of sense: it solves a problem we have now, it won't be hard to refactor out, and having this as a "real-world" example will also help make plugins better.  For example it will help us think about what configuration/credentials we need to provide to plugins, and whether we expect plugins to reuse code e.g. the cloudprovider implementations.